src/services/findService.ts

```diff
@@ -43,6 +45,20 @@ export async function fetchResources(
     }
   });
 
+  // Extract unique themes
+  const themesSet = new Set<string>();
+  const uniqueThemes: string[] = [];
+  resources.forEach((item) => {
+    if (item.theme && Array.isArray(item.theme)) {
+      item.theme.forEach((theme) => {
+        if (!themesSet.has(theme)) {
+          uniqueThemes.push(theme);
+          themesSet.add(theme);
+        }
+      });
+    }
+  });
+
```

Comment:
What you've got here works fine, but you could simplify it a tiny bit.
A `Set` doesn't have any duplicated elements, by definition, so you can call `.add` as much as you want without worrying about whether something is already in there.
And then you can just convert the Set to an Array afterwards, without needing to declare `uniqueThemes` up-front
```suggestion
  const themesSet = new Set<string>();
  resources.forEach((item) => {
    if (item.theme && Array.isArray(item.theme)) {
      item.theme.forEach((theme) => {
          themesSet.add(theme);
      });
    }
  });
  const uniqueThemes = Array.from(themesSet)
```